### PR TITLE
fixes #5141 - User unable to access children taxonomies on switcher

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
@eLobato, please confirm that your issue is regarding the org/location switcher, and this PR fixes it.

**Comment by isratrade** [comment]
[test]

**Comment by dLobatog** [comment]
Yes it is, CI is down but I tested it locally. Merged as https://github.com/theforeman/foreman/commit/53516db7c7646ec76ce449793a4bc8048d48b787 , thanks @isratrade

**Comment by domcleal** [comment]
This broke develop, please wait until CI's back up in future: http://ci.theforeman.org/job/test_develop/622/

<pre>
UsergroupTest.test_0004_hosts should be retrieved from recursive/complex usergroup definitions
&lt;[]> expected but was
&lt;[#&lt;Host::Managed id: 1073012888, name: "h1.someware.com", ip: "2.3.4.10", last_compile: nil, last_freshcheck: nil, last_report: nil, updated_at: "2014-04-14 11:55:02", source_file_id: nil, created_at: "2014-04-14 11:55:02", mac: "22:33:44:55:66:01", root_pass: nil, serial: nil, puppet_status: 0, domain_id: 22495316, architecture_id: 501905019, operatingsystem_id: 1073012828, environment_id: 334344675, subnet_id: 980190962, ptable_id: 980190962, medium_id: nil, build: false, comment: nil, disk: nil, installed_at: nil, model_id: nil, hostgroup_id: nil, owner_id: 980191116, owner_type: "User", enabled: true, puppet_ca_proxy_id: 182953976, managed: false, use_image: nil, image_file: nil, uuid: nil, compute_resource_id: nil, puppet_proxy_id: 182953976, certname: nil, image_id: nil, organization_id: nil, location_id: nil, type: "Host::Managed", otp: nil, realm_id: nil, compute_profile_id: nil, provision_method: nil>]>. (test_0004_hosts should be retrieved from recursive/complex usergroup definitions)
/var/lib/workspace/workspace/test_develop/database/mysql/ruby/1.9.3/test/unit/usergroup_test.rb:65
</pre>

File: test/unit/user_test.rb [hunk 1]
```diff
@@ -467,6 +467,26 @@ def setup_user operation
     assert users(:one).valid?
   end
 
+  test "return location and child ids for non-admin user" do
+    user = users(:one)
+    User.current = user
+    location = taxonomies(:location1)
+    Location.current = location
```
Comment (domcleal) [inline test/unit/user_test.rb:474]: These thread-local changes are probably not being reset after the test execution, hence knock-on failures.  I'd suggest reimplementing as_user for taxonomy or adding teardown methods in test_helper to ensure they're set back to nil.

File: test/unit/user_test.rb [hunk 2]
```diff
@@ -467,6 +467,26 @@ def setup_user operation
     assert users(:one).valid?
   end
 
+  test "return location and child ids for non-admin user" do
+    user = users(:one)
+    User.current = user
```
Comment (domcleal) [inline test/unit/user_test.rb:472]: use as_user

**Pull request closed without merging**